### PR TITLE
Fix print for map resources in Ruby

### PR DESCRIPTION
**Pull request by saicheems**
Also fix a missing ')' bug.
The '<<' was changed to '+' because the key string is immutable.

### CODE REVIEW

**Comment by saicheems** [comment]
Well it's not so strange, see the Java counterpart as an example:
https://github.com/googleapis/toolkit/blob/master/src/test/java/com/google/api/codegen/testdata/discoveries/java/java_compute.v1.json.baseline#L372

Each response in contains a map, in this case we call `fetch_all` to grab all the responses, which returns a list of maps. Then we iterate in a loop over each map to print its contents.

I tried putting `pretty_generate` back now that the bug is fixed... it doesn't crash anymore, but it doesn't actually pretty print either. It behaves the same as `to_json` does at the moment :/

I did try to poke around the client library to try to get the old behavior back, but I couldn't get it to work. It seems there can be issues with supporting the interface with the native JSON library's `pretty_generate` while also having custom `to_json` functions.

**Comment by saicheems** [comment]
Nvm, I see what you're saying - Ruby generally pulls out the page streaming resource in the `fetch_all` call. The reason it doesn't do so here I think is a result of this line:

https://github.com/google/google-api-ruby-client/blob/c645e7d2f893dc5165ce6354b07f147387bce109/lib/google/apis/core/base_service.rb#L72

Since the result of a list function that returns an object is a map, the generator yields the full object instead of a number of map entries like how arrays are split apart (see line 66).

**Comment by saicheems** [comment]
I filed an issue here:

https://github.com/google/google-api-ruby-client/issues/584

File: src/test/java/com/google/api/codegen/testdata/discoveries/ruby/ruby_compute.v1.json.baseline [hunk 1]
```diff
@@ -29,9 +29,11 @@ items = service.fetch_all do |token|
...
 end
 
-items.each do |name, addresses_scoped_list|
-  # TODO: Change code below to process each (name, addresses_scoped_list) pair:
-  puts String(name << " => " << addresses_scoped_list.to_json
+items.each do |item|
+  item.each do |name, addresses_scoped_list|
+    # TODO: Change code below to process each (name, addresses_scoped_list) pair:
+    puts String(name) + " => " + addresses_scoped_list.to_json
+  end
```
Comment (tcoffee-google) [inline src/test/java/com/google/api/codegen/testdata/discoveries/ruby/ruby_compute.v1.json.baseline]: Given the way the library currently handles maps, it seems to me that to be conceptually right (and avoid confusing the user), the variable names for this case will need to change---something like: `items` => `page`, `item` => `items`.

Comment (saicheems) [inline src/test/java/com/google/api/codegen/testdata/discoveries/ruby/ruby_compute.v1.json.baseline]: I'm not sure if `page` is the right name either, the collection returned by `fetch_all` is the list of `items` pulled from the response, not the page. I think that `items` is still the correct name.

https://github.com/saicheems/toolkit/blob/master/src/test/java/com/google/api/codegen/testdata/discoveries/compute.v1.json#L199

Here's the discovery, the items field is what's collected by `fetch_all`.

**Comment by saicheems** [comment]
Note: This depends on the next client release with https://github.com/google/google-api-ruby-client/pull/586

**Comment by saicheems** [comment]
Updated, PTAL

**Pull request merged**